File: LeetCode_Solutions/LeetCode_Solutions/SumOfTwoIntegers_371.m
//
//  SumOfTwoIntegers_371.m
//  LeetCode_Solutions
//
//  Created by Huni on 11/04/2017.
//  Copyright © 2017 KnightJoker. All rights reserved.
//



#import "SumOfTwoIntegers_371.h"

@implementation SumOfTwoIntegers_371

- (NSInteger)getSum:(NSInteger)a otherNum:(NSInteger)b {

    if(a == 0) return b;
    
    NSInteger x = a ^ b;
    NSInteger c = (a & b) << 1;
    return [self getSum:c otherNum:x];
}
@end

File: LeetCode_Swift/LeetCode_Swift/SumOfTwoIntegers_371.swift
//
//  SumOfTwoIntegers_371.swift
//  LeetCode_Swift
//
//  Created by Huni on 28/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class SumOfTwoIntegers_371 {

    func getSum(_ a: Int, _ b: Int) -> Int {
        if a == 0 {
            return b
        }
        let x = a ^ b
        let c = (a & b) << 1
        return self.getSum(_: c ,_: x)
    }
}



